src/workerd/api/queue.h

```diff
@@ -108,24 +108,48 @@ struct IncomingQueueMessage {
   };
 };
 
+struct QueueRetryBatch {
+  bool retry;
+  jsg::Optional<int> delaySeconds;
+  JSG_STRUCT(retry, delaySeconds);
+};
+
+struct QueueRetryMessage {
+  kj::String msgId;
+  jsg::Optional<int> delaySeconds;
+  JSG_STRUCT(msgId, delaySeconds);
+};
+
 struct QueueResponse {
   uint16_t outcome;
-  bool retryAll;
   bool ackAll;
-  kj::Array<kj::String> explicitRetries;
+  QueueRetryBatch retryBatch;
   kj::Array<kj::String> explicitAcks;
-  JSG_STRUCT(outcome, retryAll, ackAll, explicitRetries, explicitAcks);
+  kj::Array<QueueRetryMessage> retryMessages;
+  JSG_STRUCT(outcome, ackAll, retryBatch, explicitAcks, retryMessages);
 };
 
 // Internal-only representation used to accumulate the results of a queue event.
 
 struct QueueEventResult {
-  bool retryAll = false;
+  struct RetryOptions {
+    jsg::Optional<int> delaySeconds;
+  };
+  struct RetryBatch {
+    bool retry;
+    jsg::Optional<int> delaySeconds;
+  };
+  RetryBatch retryBatch = {.retry = false};
   bool ackAll = false;
-  kj::HashSet<kj::String> explicitRetries;
+  kj::HashMap<kj::String, RetryOptions> retries;
```

Comment:
Are these `retryMessages`, `retryBatches` or both? Does it make sense to make this a little more explicit?